android/src/main/java/com/swmansion/reanimated/NodesManager.java

```diff
@@ -390,17 +393,43 @@ public void synchronouslyUpdateUIProps(int viewTag, ReadableMap uiProps) {
   }
 
   public String obtainProp(int viewTag, String propName) {
-    View view = mUIManager.resolveView(viewTag);
-    String result =
-        "error: unknown propName " + propName + ", currently supported: opacity, zIndex";
-    if (propName.equals("opacity")) {
-      Float opacity = view.getAlpha();
-      result = Float.toString(opacity);
-    } else if (propName.equals("zIndex")) {
-      Float zIndex = view.getElevation();
-      result = Float.toString(zIndex);
+    View view;
+    try {
+      view = mUIManager.resolveView(viewTag);
+    } catch (Exception e) {
+      throw new IllegalStateException("[Reanimated] Unable to resolve view");
+    }
+
+    switch (propName) {
+      case "opacity":
+        return Float.toString(view.getAlpha());
+      case "zIndex":
+        return Float.toString(view.getElevation());
+      case "width":
+        float width = PixelUtil.toDIPFromPixel(view.getWidth());
+        return Float.toString(width);
```

Comment:
Honestly I don't think we need to introduce intermediate variables here, what do you think?
```suggestion
        return Float.toString(PixelUtil.toDIPFromPixel(view.getWidth()));
```